include/onnxruntime/core/common/const_pointer_container.h

```diff
@@ -64,6 +64,10 @@ class ConstPointerContainer {
   explicit ConstPointerContainer(const Container& data) noexcept : data_(data) {}
 
   size_t size() const noexcept { return data_.size(); }
+  bool empty() const noexcept { return data_.empty(); }
+
+  ConstIterator cbegin() const noexcept { return ConstIterator(data_.cbegin()); }
+  ConstIterator cend() const noexcept { return ConstIterator(data_.cend()); }
 
   ConstIterator begin() const noexcept { return ConstIterator(data_.cbegin()); }
```

Comment:
Do we need this to be calling data_.cbegin()/cend(), given that we have the cbegin/cend methods now?

Reply:
Given this wrapper is just providing constness, it doesn't matter which one you call. This matches behaviour for a const std::vector where you get std::vector::const_vector regardless of whether you call begin() or cbegin(). 

In general, prefer using cbegin/cend unless you're going to modify something via the iterator as that advertises the intent more clearly, so adding cbegin and cend supports that pattern.